lib/claim_letters/claim_letter_downloader.rb

```diff
@@ -27,16 +27,19 @@ def get_letters
     end
 
     def get_letter(document_id)
-      verify_letter_in_folder(document_id)
+      letter_details = get_letter_details(document_id)
+      raise Common::Exceptions::RecordNotFound, document_id if letter_details.blank?
+
+      filename = filename_with_date('ClaimLetter', letter_details[0][:received_at])
```

Comment:
(See comment below). If we were to use a `find` instead of a `select` in `get_letter_details` we could avoid having to index this array that's either 1 or 0 elements long every time. If we return either the letter or `nil` when we get the letter details, a simple `letter_details.nil?` would suffice in the exception raising code above.